src/main/java/org/broadinstitute/hellbender/tools/walkers/mutect/SomaticGenotypingEngine.java

```diff
@@ -98,15 +96,19 @@ public CalledHaplotypes callMutations(
 
         // update the haplotypes so we're ready to call, getting the ordered list of positions on the reference
         // that carry events among the haplotypes
-        final List<Integer> startPosKeySet = decomposeHaplotypesIntoVariantContexts(haplotypes, assemblyResultSet.getFullReferenceWithPadding(), assemblyResultSet.getPaddedReferenceLoc(), NO_GIVEN_ALLELES).stream()
+        final List<Integer> startPosKeySet = decomposeHaplotypesIntoVariantContexts(haplotypes, assemblyResultSet.getFullReferenceWithPadding(), assemblyResultSet.getPaddedReferenceLoc(), givenAlleles).stream()
```

Comment:
Inside `AssemblyBasedCallerGenotypingEngine::decomposeHaplotypesIntoVariantContexts` (the whole method copied below), we call `startPosKeySet.clear()`, which, as far as I can tell, ensures that the only GAA variants go through and that's not what we want. Since we've already inserted the GGA alleles into haplotypes already, it seems that we can remove the `if (inGGAMode)` block entirely.

```java
    protected TreeSet<Integer> decomposeHaplotypesIntoVariantContexts(final List<Haplotype> haplotypes,
                                                                      final byte[] ref,
                                                                      final SimpleInterval refLoc,
                                                                      final List<VariantContext> activeAllelesToGenotype) {
        final boolean inGGAMode = ! activeAllelesToGenotype.isEmpty();

        // Using the cigar from each called haplotype figure out what events need to be written out in a VCF file
        // IMPORTANT NOTE: This needs to be done even in GGA mode, as this method call has the side effect of setting the
        // event maps in the Haplotype objects!
        final TreeSet<Integer> startPosKeySet = EventMap.buildEventMapsForHaplotypes(haplotypes, ref, refLoc, configuration.debug);

        if ( inGGAMode ) {
            startPosKeySet.clear(); // <<<< (emphasis by Takuto)
            for( final VariantContext compVC : activeAllelesToGenotype ) {
                startPosKeySet.add(compVC.getStart());
            }
        }

        return startPosKeySet;
    }
```

Reply:
Alternatively we can give this method an empty list instead of `givenAlleles` as you do several lines below with `getVCsAtThisLocation`

Reply:
Wow.  Really good catch.  I thought I had tracked down all the places where GGA mode overrides discovery alleles.  Code review at its finest.